File: CSharp/Bases/ExtensionBase.cs
using System;

namespace N2f
{
	/// <summary>
	/// Abstract base for Extension receivers.
	/// </summary>
	public abstract class ExtensionBase
	{
		/// <summary>
		/// Method to initialize an extension.
		/// </summary>
		/// <param name="N2f">The current executing instance.</param>
		abstract public void Initialize(N2f N2f);
	}
}

File: PHP/N2f/Core/BaseClasses/Extension.base.php
<?php

	namespace N2f;

	/**
	 * Abstract for extension 'base' classes.
	 *
	 * Abstract which defines the Initialize() method used
	 * by N2f to initialize extensions with injection.
	 *
	 * @version 1.0
	 * @author Andrew Male
	 * @copyright 2014-2016 Zibings.com
	 * @package N2F
	 */
	abstract class ExtensionBase {
		/**
		 * Method to receive initialization pulse from
		 * the N2f instance.
		 * 
		 * @param \N2f\N2f $N2f The N2f instance initializing the extension.
		 */
		abstract public function Initialize(N2f &$N2f);
	}
